app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
@agrare, been trying to solve this the best way, but while doing it got back a bit of context and turns out we'll always have the need to do two call to `find_by`, in case it's not a Server, e.g. if it's a MiddlewareDeployment we need to look it up, get the server_id from the MiddlewareDeployment and look up the actual server to get the name.

So, for what we have now, where the operations only affect a Deployment or a Server, as it is seems to fit. For the future will need to support different MW resources.

Let me know if you understand it and sorry for the confusion.